WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift

```diff
@@ -4,21 +4,6 @@ import protocol Yosemite.POSItem
 import protocol Yosemite.POSItemProvider
 
 final class ItemListViewModel: ObservableObject {
-    enum ItemListState {
-        // TODO:
-        // Differentiate between loading on entering POS mode and reloading, as the
-        // screens will be different:
-        // https://github.com/woocommerce/woocommerce-ios/issues/13286
-        case loading
-        // TODO:
-        // Differentiate between loaded with products vs with no products vs with no eligible products
-        // https://github.com/woocommerce/woocommerce-ios/issues/12815
-        // https://github.com/woocommerce/woocommerce-ios/issues/12816
-        case loaded
-        // TODO:
-        // https://github.com/woocommerce/woocommerce-ios/issues/12846
-        case error
-    }
 
     @Published private(set) var items: [POSItem] = []
```

Comment:
Are you referring to `@Published private(set) var items: [POSItem] = []`? There's a reference to this property from the dashboard when we sync the order, so I can't remove it 😅  :

```
private extension PointOfSaleDashboardViewModel {
    func startSyncingOrder(cartItems: [CartItem]) {
        totalsViewModel.startSyncingOrder(with: cartItems,
                                          allItems: itemListViewModel.items)
    }
}
```
I'm not quite sure to understand why we need to pass all products when synchronizing, I think could be just because we're reusing `app` logic for the moment that needs those. Maybe @bozidarsevo has more context here.

Reply:
Got it, I haven't seen that it was used this way. Ideally, we would rely on the state since it also contains items. 

We could have something like:

```
var items: [POSItem] {
   switch state {
        case let .success(items)
            return items
         default
             return []
   }
}
``

Reply:
> all products

yep, we need to have list of all products for the order syncing logic